override/blocks/lists.js

```diff
@@ -50,7 +50,7 @@ Blockly.Blocks['lists_create_with'] = {
                 var input = this.appendValueInput('ADD' + i);
                 if (i == 0) {
                     input.appendField(new Blockly.FieldArrayLength(this.itemCount_, (newValue) => {
-                        if (this.itemCount_ !== newValue) {
+                        if (typeof newValue === 'number' && this.itemCount_ !== newValue) {
```

Comment:
Maybe the FieldArrayLength should make sure to only set number values? Or if it is not a number, parsing int to one would probably work